### PR TITLE
23.3 osx-arm64 build

### PR DESCRIPTION
**Pull request by mikemhenry**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mikemhenry** [comment]
@dacase is `nab2c` still needed?

File: recipe/build.sh [hunk 1]
```diff
@@ -52,18 +52,86 @@ if [[ "$target_platform" == osx* ]]; then
...
+    @@ -214,7 +214,7 @@
+     if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
+     	add_flags(CXX -Wall -Wno-unused-function)
+     	
+    -	list(APPEND OPT_CXXFLAGS "-mtune=native")
+    +	#list(APPEND OPT_CXXFLAGS "-mtune=native")
+     	
+     	if(CROSSCOMPILE)
+     		set(CMAKE_CXX_COMPILER_TARGET ${TARGET_TRIPLE})
+EOF
```
Comment (jaimergp) [inline recipe/build.sh:83]: This should probably be its own patch file, with a selector on meta.yaml's `source.patches` entry. Doing this you are also adding extra leading space, which might be affecting the patch? Not sure on that one.

File: recipe/build.sh [hunk 2]
```diff
@@ -52,18 +52,86 @@ if [[ "$target_platform" == osx* ]]; then
...
+
+    make
+    make install
+    # Now build the package
+    mkdir -p build
+    cd build || exit
+    cmake ${CMAKE_ARGS} ${SRC_DIR} ${CMAKE_FLAGS} \
+        -DBUILD_HOST_TOOLS=FALSE \
+        -DUSE_HOST_TOOLS=TRUE \
+        -DHOST_TOOLS_DIR="${BUILD_PREFIX}/amber_host_tools" \
```
Comment (jaimergp) [inline recipe/build.sh:108]: This is the only difference between the two invokation types, so I'd put this in a variable like `ARGS_FOR_ARM64` or something, which would be empty when not cross-compiling.

File: recipe/build.sh (hunk 1 above)
Comment (mikemhenry) [reply recipe/build.sh:83]: Good point, I may have to remake this patch anyway since the source in 23 may be different

**Comment by mikemhenry** [comment]
getting closer!
```
2023-06-21T22:29:18.1547860Z Scanning dependencies of target xc
2023-06-21T22:29:18.1962400Z [ 10%] Building Fortran object AmberTools/src/quick/src/libxc/CMakeFiles/xc.dir/libxc_funcs.f90.o
2023-06-21T22:29:18.2224840Z f951: Error: unknown value ‘native’ for ‘-mtune’
2023-06-21T22:29:18.2228130Z f951: note: valid arguments are: cortex-a34 cortex-a35 cortex-a53 cortex-a57 cortex-a72 cortex-a73 thunderx thunderxt88p1 thunderxt88 octeontx octeontx81 octeontx83 thunderxt81 thunderxt83 ampere1 emag xgene1 falkor qdf24xx exynos-m1 phecda thunderx2t99p1 vulcan thunderx2t99 cortex-a55 cortex-a75 cortex-a76 cortex-a76ae cortex-a77 cortex-a78 cortex-a78ae cortex-a78c cortex-a65 cortex-a65ae cortex-x1 ares neoverse-n1 neoverse-e1 octeontx2 octeontx2t98 octeontx2t96 octeontx2t93 octeontx2f95 octeontx2f95n octeontx2f95mm a64fx tsv110 thunderx3t110 zeus neoverse-v1 neoverse-512tvb saphira cortex-a57.cortex-a53 cortex-a72.cortex-a53 cortex-a73.cortex-a35 cortex-a73.cortex-a53 cortex-a75.cortex-a55 cortex-a76.cortex-a55 cortex-r82 cortex-a510 cortex-a710 cortex-x2 neoverse-n2 demeter generic
2023-06-21T22:29:18.2240050Z make[2]: *** [AmberTools/src/quick/src/libxc/CMakeFiles/xc.dir/build.make:3043: AmberTools/src/quick/src/libxc/CMakeFiles/xc.dir/libxc_funcs.f90.o] Error 1
2023-06-21T22:29:18.2243460Z make[1]: *** [CMakeFiles/Makefile2:4594: AmberTools/src/quick/src/libxc/CMakeFiles/xc.dir/all] Error 2
2023-06-21T22:29:18.2247740Z make: *** [Makefile:156: all] Error 2
```

Looks like I need to play with the flags passed to FORTRAN

**Comment by ghutchis** [comment]
I'd remove the `-mtune=native` completely. Getting a working version would be great, regardless of CPU tuning.

**Comment by mikemhenry** [comment]
@ghutchis I've got an ad-hoc patch here https://github.com/conda-forge/ambertools-feedstock/pull/116/files#diff-d7075654874cb08007a21aaab3ecd4b3453a9087e7505d034d548b8938b599bcR62-R83 

So I just must have missed one

**Comment by mikemhenry** [comment]
Patching this out is turning into more work... I think I am going to instead try and see how I can get the cross compile to work instead

**Comment by mikemhenry** [comment]
@jaimergp I think I need to add something here https://github.com/conda-forge/ambertools-feedstock/pull/116/files#diff-d7075654874cb08007a21aaab3ecd4b3453a9087e7505d034d548b8938b599bcR61-R69 to build the host tools using x86 arch and not arm

**Comment by mikemhenry** [comment]
I've got some clean up to do, but it might be working now:
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=734895&view=artifacts&pathAsName=false&type=publishedArtifacts

@mattwthompson do you mind testing this build?

**Review by mattwthompson — APPROVED** [review]
I created a local environment, grabbed a `.conda` file from artifacts, and got relevant toolkit and Interchange tests passing.

The only quirk was needing to install some deps like `arpack`, `fftw`, etc. manually. I'm attributing this to local installation since that's not been a problem before

**Comment by hadim** [comment]
Thanks @mikemhenry, for the daunting effort on this!

**Comment by mikemhenry** [comment]
I've also had tests pass locally :) I just didn't belive it haha. After getting everything unzipped, this is how I installed it `micromamba create -n openff-amber23 openff-toolkit osx-arm64/ambertools-23.3-py310hd4daa78_1.conda` and it pulled in `arpack                                3.7.0  h58ebc17_2               conda-forge/osx-arm64        200kB` and `fftw                                 3.3.10  nompi_h3046061_108       conda-forge/osx-arm64        785kB`


I will remove some of the debugging stuff in this PR, merge it, and pour a glass of scotch

**Comment by isuruf** [comment]
@mikemhenry, please keep my commits with my author info next time.

**Comment by mikemhenry** [comment]
@isuruf sorry! I didn't realize I lost them! I just looked back in the commit history and noticed that when I made the 2nd PR from https://github.com/conda-forge/ambertools-feedstock/pull/80 I started over using the 23 build version and your commits were from the 22 build version! I am really sorry, you are talking about these, right?

e260850480f6c1ba0696c31a68ed94802715ad99
561bc0a4149d74316fdce08adc49b7ce4d51631d
1b60d3a70e8b322b28046247bcc5b22b68e3b6d2
e260850480f6c1ba0696c31a68ed94802715ad99

**Comment by isuruf** [comment]
Yes, those. It's fine for the first time. Let's remember to not do that next time.
